modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java

```diff
@@ -838,4 +842,54 @@ public void onLeaderStart(long term) {
             listener.onLeaderStart();
         }
     }
+
+    /** Returns raft group state accessor. */
+    public <G extends ReplicationGroupId, L> RaftGroupStateProvider<G, L> getGroupStateProvider(
+            Class<G> groupType,
+            Class<L> groupListenerType
+    ) {
+        return new StateProviderImpl<>(groupType, groupListenerType, nodes);
+    }
+
+    private static class StateProviderImpl<G extends ReplicationGroupId, L> implements RaftGroupStateProvider<G, L> {
+
+        private final Class<G> groupType;
+
+        private final Class<L> listenerType;
+
+        private final Map<RaftNodeId, RaftGroupService> nodes;
+
+        StateProviderImpl(
+                Class<G> groupType,
+                Class<L> listenerType,
+                Map<RaftNodeId, RaftGroupService> nodes
+        ) {
+            this.groupType = groupType;
+            this.listenerType = listenerType;
+            this.nodes = nodes;
+        }
+
+        @Override
+        public <V> Map<G, V> getState(Function<L, V> stateFunc) {
+            Map<G, V> result = new HashMap<>();
+
+            for (Map.Entry<RaftNodeId, RaftGroupService> entry : nodes.entrySet()) {
+                RaftNodeId nodeId = entry.getKey();
+                ReplicationGroupId replicationGroupId = nodeId.groupId();
```

Comment:
All these `nodes` belong to table partitions, so I think have using a loop here is ok.